clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -35,8 +36,9 @@
                                                                      INT32,
                                                                      "A user-supplied integer value that will be passed back with the response"),
                                                            new Field("client_id",
```

Comment:
Per David's report in KAFKA-3088, he was sending an "empty" client-id. We could try and reproduce it with Bruce, but I'm basically unclear on how we could have ever parsed out a `null` since the `Type.read` for `String` should never have worked right?